purple-pill/js/controllers/fontexportController.js

```diff
@@ -2,6 +2,72 @@ app.controller('fontexportController', ['$scope', '$http', 'sharedScope', '$time
 function($scope, $http, sharedScope, $timeout) {
     $scope.data = sharedScope.data;
 
+    var ExportObject = (function() {
+        function ExportObject(instance, fileFormat) {
+            var retval;
+            this.fileFormat = fileFormat || "UFO";
+            this.instance = instance;
+
+            if (this.fileFormat == "UFO"){
+                getGenerator = $scope.data.stateful.project.getUFOExportGenerator.bind($scope.data.stateful.project);
+//TODO:     } else if (this.fileFormat == "OTF"){
+//TODO:         getGenerator = $scope.data.stateful.project.getOTFExportGenerator.bind($scope.data.stateful.project);
+            }
+            retval = getGenerator(
+                instance.name
+              , this.getFileName()
+              , /* precision: */ -1
+            );
+            this.generator = retval[0];
+            this.io = retval[1];
+        }
+
+        var _p = ExportObject.prototype;
+        _p.constructor = ExportObject;
+
+        _p.getFileName = function() {
+            switch (this.fileFormat){
+                case "UFO":
+                    return this.instance.displayName + ".ufo";
+                case "OTF":
+                    return this.instance.displayName + ".otf";
+            }
+        };
+
+        return ExportObject;
+    })();
+
+    var ProgressBar = (function() {
+        function ProgressBar(barElementId, labelElementId, updateDelay) {
```

Comment:
Also note, that a Progressbar would be better implemented as an AngularJS module. But doing that will be easier when Jeroens port has landet, so don't bother now.